src/plugins/mer/mersettings.cpp

```diff
@@ -123,6 +165,49 @@ QMap<QString, MerEmulatorDeviceModel> MerSettings::deviceModelsRead(const Utils:
     return result;
 }
 
+bool MerSettings::isDeviceModelStored(const MerEmulatorDeviceModel &model)
+{
+    Q_ASSERT(s_instance);
+
+    if (!s_instance->m_deviceModels.contains(model.name()))
+        return false;
+
+    return s_instance->m_deviceModels.value(model.name()) == model;
+}
+
+void MerSettings::setDeviceModels(const QMap<QString, MerEmulatorDeviceModel> &deviceModels)
+{
+    Q_ASSERT(s_instance);
+
+    if (s_instance->m_deviceModels == deviceModels)
+        return;
+
+    s_instance->m_deviceModels = deviceModels;
+    emit s_instance->deviceModelsChanged();
+}
+
+void MerSettings::deviceModelsWrite(const Utils::FileName &fileName,
+                                           const QMap<QString, MerEmulatorDeviceModel> &deviceModels)
+{
+    PersistentSettingsWriter writer(fileName, QLatin1String("QtCreatorMersdk-user-defined-device-models"));
```

Comment:
The doctype should be the same as for the SDK-provided device models

Reply:
Fixed.